packages/list/ListItem.js

```diff
@@ -72,14 +72,16 @@ export default class ListItem extends Component {
       ...otherProps
     } = this.props;
 
+    const SemanticListItem = this.props.href ? 'a' : 'li';
```

Comment:
I do agree with this change, but I think there should be a way to programatically change routes. Things like React Router and SPA architecture require this. Semantically we will want an anchor tag, but no href, correct?

Reply:
I think so? I'm having a hard time with the layers of abstraction here. Below is an example of how I would imagine it working while using [`next`](https://github.com/zeit/next.js) and its routing logic via `next/link`.

```jsx
import Link from 'next/link';
import List, { 
  ListItem,
  ListItemGraphic,
  ListItemText,
} from '@material/react-list';

const NavList = [
  {
    href: '/',
    name: 'Home',
    icon: 'home',
  }, {
    href: '/info',
    name: 'Info',
    icon: 'info',
  }
];

export default () => (
  <List wrapFocus tag="nav">
    {NavList.map(({ href, name, icon }, index) => (
      <Link key={index} href={href} prefetch passHref>
        <ListItem key={index} tag="a">
          <ListItemGraphic graphic={<MaterialIcon icon={icon} />} />
          <ListItemText primaryText={name} />
        </ListItem>
      </Link>
      // With react-router-dom it would be pretty similar...
      // <Link to={href}>
      //   <ListItem ...>
      //     <ListItemGraphic ... />
      //     <ListItemText ... />
      //   </ListItem>
      // </Link>
    ))}
  </List>
);
```

Reply:
What if people wanted to do 
```
<ListItem key={index} tag="a" onClick={this.goToAnotherRoute}>
          <ListItemGraphic graphic={<MaterialIcon icon={icon} />} />
          <ListItemText primaryText={name} />
</ListItem>
```

All I'm saying is, we should remove `const SemanticListItem = this.props.href ? 'a' : 'li';`, in favor of the `Tag` property.

Reply:
> What if people wanted to do
> 
> ```
> <ListItem key={index} tag="a" onClick={this.goToAnotherRoute}>
>           <ListItemGraphic graphic={<MaterialIcon icon={icon} />} />
>           <ListItemText primaryText={name} />
> </ListItem>
> ```
> All I'm saying is, we should remove `const SemanticListItem = this.props.href ? 'a' : 'li';`, in favor of the `Tag` property.

Got it. I've updated `ListItem` accordingly and updated the README.

Another thing to consider: When using router frameworks that nest elements within `<Link></Link>`, I encountered problems where the error output is something to the effect of `Cannot set property 'aria-selected' of undefined`, and when using the `selectedIndex` prop, it’s unable to set it on the index specified. Would this be fixed if users extended the `Link` component to pass the necessary props to its children (i.e., `ListItem`)? Or how else can we make it safe to nest `ListItem`?

I have the beginnings of that in [my project](https://github.com/4cm4k1/website/blob/master/components/Link.jsx), though currently using RMWC because of the crashing issues here, shown here:

```jsx
<Link {...props}>
  <ListItem tag="a" activated={activated}>
    {children}
  </ListItem>
</Link>
```